adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/FeatureRDD.scala

```diff
@@ -284,14 +285,36 @@ case class DatasetBoundFeatureRDD private[rdd] (
                              blockSize: Int = 128 * 1024 * 1024,
                              pageSize: Int = 1 * 1024 * 1024,
                              compressCodec: CompressionCodecName = CompressionCodecName.GZIP,
-                             disableDictionaryEncoding: Boolean = false) {
-    log.warn("Saving directly as Parquet from SQL. Options other than compression codec are ignored.")
-    dataset.toDF()
-      .write
-      .format("parquet")
-      .option("spark.sql.parquet.compression.codec", compressCodec.toString.toLowerCase())
-      .save(filePath)
-    saveMetadata(filePath)
+                             disableDictionaryEncoding: Boolean = false,
+                             enableHivePartitioning: Boolean,
+                             hivePartitioningGenomicBinSize: Int) {
+
+    if (!enableHivePartitioning) {
+      log.warn("Saving directly as Parquet from SQL. Options other than compression codec are ignored.")
+      dataset.toDF()
+        .write
+        .format("parquet")
+        .option("spark.sql.parquet.compression.codec", compressCodec.toString.toLowerCase())
+        .save(filePath)
+      saveMetadata(filePath)
+    } else {
+      log.warn("Saving directly as Hive-partitioned Parquet from SQL. " +
+        "Options other than compression codec are ignored.")
+      import org.apache.spark.sql.functions._
+      val genomicBinSize = hivePartitioningGenomicBinSize
+      val df = dataset.toDF()
+
+      df.withColumn("posBin", floor(df("start") / genomicBinSize))
+        .write
+        .partitionBy("contigName", "posBin")
+        .format("parquet")
+        .option("spark.sql.parquet.compression.codec", compressCodec.toString.toLowerCase())
+        .save(filePath)
+
+      val hiveFlagFile = new File(filePath, "_hivePartitioned")
```

Comment:
These will also need to be rewritten to use either the Hadoop FileSystem API or the java.nio libraries.